### ISSUE TITLE
R 4.0 raw string support

### ISSUE THREAD

**Issue by renkun-ken** [jimhester/lintr#484]
R 4.0 introduces raw string in the form of `r"(hello, "world")"` (see https://stat.ethz.ch/R-manual/R-devel/library/base/html/Quotes.html).

For the following code:

```r
x <- r"(hello, "world")"
```

The lintr runs without a problem.

For the following incorrect code:

```r
text <- r"c:\wind"
```

`parse()` says:

```
Error in parse("r4.R") : malformed raw string literal at line 4
```

but `lintr::lint()` ends up with an error:

```
Error in source_file$lines[[line_number]] : subscript out of bounds
Backtrace:
1: structure(list(filename = filename, line_number = as.integer(line_number), 
2: Lint(filename = source_file$filename, line_number = line_number, 
3: value[[3L]](cond)
4: tryCatchOne(expr, names, parentenv, handlers[[1L]])
5: tryCatchList(expr, classes, parentenv, handlers)
6: tryCatch(source_file$parsed_content <- parse(text = source_file$content, 
7: get_source_file(source_file, error = lint_error)
8: get_source_expressions(filename)
9: lintr::lint("r4.R")
```

**Comment by AshesITR** [comment]
I can no longer reproduce in R 4.0.3
Closing as it seems to have been fixed in [R 4.0.1](https://cran.r-project.org/doc/manuals/r-devel/NEWS.html)
> Parse data for raw strings is now recorded correctly. Reported by Gabor Csardi.

```
> lintr::lint('text <- r"c:\\wind"\n')
<text>:1:1: error: malformed raw string literal at line 1

^
```
